tests/types/test_trackmate.py

```diff
@@ -0,0 +1,192 @@
+        model.addEdge(s5a, s6a, 0.0)
+        model.addEdge(s5b, s6b, 0.0)
+
+        model.addSpotTo(s7, jc.Integer(0))
+        model.addSpotTo(s8, jc.Integer(1))
+        model.addEdge(s7, s8, 0.0)
+
+        model.addSpotTo(s9, jc.Integer(0))
+        model.addSpotTo(s10, jc.Integer(1))
+        model.addEdge(s9, s10, 0.0)
+    finally:
+        model.endUpdate()
+
+    ij.object().addObject(trackMate)
+    return trackMate
+
+
+def ui_visible(ij):
+    frame = ij.ui().getDefaultUI().getApplicationFrame()
+    if isinstance(frame, jc.UIComponent):
+        frame = frame.getComponent()
+    return frame and frame.isVisible()
+
+
+@pytest.fixture(autouse=True)
+def clean_gui_elements(asserter, ij, viewer: Viewer):
+    """Fixture to remove image data from napari and ImageJ2"""
+
+    # Test pre-processing
+
+    # Test processing
+    yield
+
+    # Test post-processing
+
+    # After each test runs, clear all layers from napari
+    if viewer is not None:
+        viewer.layers.clear()
+
+    # After each test runs, clear all ImagePlus objects from ImageJ
+    if ij.legacy and ij.legacy.isActive():
+        while ij.WindowManager.getCurrentImage():
+            imp = ij.WindowManager.getCurrentImage()
+            imp.changes = False
+            imp.close()
+
+    # After each test runs, clear all displays from ImageJ2
+    while not ij.display().getDisplays().isEmpty():
+        for display in ij.display().getDisplays():
+            display.close()
+
+    # Close the UI if needed
+    if ui_visible(ij):
+        frame = ij.ui().getDefaultUI().getApplicationFrame()
+        if isinstance(frame, jc.UIComponent):
+            frame = frame.getComponent()
+        ij.thread().queue(
+            lambda: frame.dispatchEvent(
+                jc.WindowEvent(frame, jc.WindowEvent.WINDOW_CLOSING)
+            )
+        )
+        # Wait for the Frame to be hidden
+        asserter(lambda: not ui_visible(ij))
+
+
+pytest.mark.skipif(
+    not (TESTING_TRACKMATE and TESTING_LEGACY),
+    "TrackMate functionality requires ImageJ and TrackMate!",
+)
+
+
+def test_Model_to_Tracks(ij, trackMate_example):
+    # Convert the TrackMate instance into a Dataset with Rois
+    # this is what we'd convert in practice.
+    model = trackMate_example.getModel()
+    selection_model = jc.SelectionModel(model)
+    imp = ij.convert().convert(ij.py.to_java(np.zeros((10, 10))), jc.ImagePlus)
+    display_settings = jc.DisplaySettings()
+
+    hyper_stack_displayer = jc.HyperStackDisplayer(
+        model, selection_model, imp, display_settings
+    )
+    hyper_stack_displayer.render()
+
+    dataset = ij.convert().convert(imp, jc.Dataset)
+
+    # Convert the dataset's rois into its Python equivalent
+    layers = ij.py.from_java(dataset.getProperties()["rois"])
+    # Assert we receive a Tracks Layer
+    assert isinstance(layers, Tuple)
+    tracks, labels = layers
+    assert isinstance(tracks, Tracks)
+    assert isinstance(labels, Labels)
```

Comment:
Might want to test that the labels are being added.

Reply:
I'm going to leave the testing for now; the labels are created using purely TrackMate functionality. All I really think we need to do is make sure we're returning the label image, which we are doing.